hazelcast/src/main/java/com/hazelcast/core/Client.java

```diff
@@ -47,4 +48,24 @@
      * @return the type of this client
      */
     ClientType getClientType();
+
+
+    /**
+     * This method may return null depending on the client version and the client type
+     * Java client provides client name starting with 3.12
+     *
+     * @return the name of this client if provided, null otherwise
+     */
+    String getName();
+
+    /**
+     * @param attributeKey key of the attribute
+     * @return attribute value if found, null otherwise
+     */
+    String getAttribute(String attributeKey);
```

Comment:
Do we need this method? Since the one below already gives you everything you want to do with the attributes.

Reply:
@pveentjer This is sort of the requirement. It should be discussed with requirement owners on the prd. See https://hazelcast.atlassian.net/wiki/spaces/PM/pages/387252458/Attributes+for+Client

Reply:
I removed the `getAttribute` method